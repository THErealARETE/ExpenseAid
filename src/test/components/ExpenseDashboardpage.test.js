import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ExpenseDashboardPage from '../../components/ExpenseDashboardPage'; 
import expense from '../fixtures/expenses';
Enzyme.configure({
    adapter: new Adapter()
})


test('should render expense dashboard page correctly', ()=>{

    const wrapper=shallow(<ExpenseDashboardPage expenses={expense}/>)
    expect(wrapper).toMatchSnapshot()
})