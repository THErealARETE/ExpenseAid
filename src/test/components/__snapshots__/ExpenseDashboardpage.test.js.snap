// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense dashboard page correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseDashboardPage
    expenses={
      Array [
        Object {
          "amount": 200,
          "createdAt": 0,
          "description": "Gem",
          "id": "1",
          "note": "",
        },
        Object {
          "amount": 300,
          "createdAt": -345600000,
          "description": "wezzy",
          "id": "2",
          "note": "everything stem ",
        },
        Object {
          "amount": 400,
          "createdAt": 345600000,
          "description": "arm",
          "id": "3",
          "note": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(ExpenseListFilters) />,
        <Connect(ExpenseList) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(ExpenseListFilters) />,
          <Connect(ExpenseList) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
